src/loop.c

```diff
@@ -262,10 +262,14 @@ handles_walk_cb(uv_handle_t* handle, void* arg)
 
     handles = arg;
     obj = handle->data;
+    printf("%p\n", &HandleType);
+    printf("%s\n", Py_TYPE(obj)->tp_name);
 
-    ASSERT(obj);
-
-    PyList_Append(handles, obj);
+    if (IS_PYUV_HANDLE(obj)) {
+        if (!PyErr_Occurred()) {
```

Comment:
There's pretty much no point in adding other items to the list if an exception has occured, as we'll have to return null (and decref the list) in `Loop_handles_get` (otherwise Python would throw SystemError when control returns to Python code as it expects NULL if an exception has occurred.)